internal/ignition/ignition_test.go

```diff
@@ -0,0 +1,109 @@
+package ignition
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	config_31 "github.com/coreos/ignition/v2/config/v3_1"
+	config_31_types "github.com/coreos/ignition/v2/config/v3_1/types"
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	bmh_v1alpha1 "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+var _ = Describe("Ignition", func() {
+	var (
+		workDir     string
+		err         error
+		examplePath string
+		log         = logrus.New()
+		cluster     *common.Cluster
+		file        *config_31_types.File
+		bmh         *bmh_v1alpha1.BareMetalHost
+	)
+
+	BeforeEach(func() {
+		var err1 error
+		workDir, err1 = ioutil.TempDir("", "assisted-install-test-")
+		Expect(err1).NotTo(HaveOccurred())
+		installerCacheDir := filepath.Join(workDir, "installercache")
+		examplePath = filepath.Join(workDir, "example1.ign")
+		clusterID := strfmt.UUID(uuid.New().String())
+		cluster = &common.Cluster{
+			Cluster: models.Cluster{
+				ID: &clusterID,
+				Hosts: []*models.Host{
+					{
+						Inventory:         inventory1,
+						RequestedHostname: "example1",
+						Role:              models.HostRoleMaster,
+					},
+				},
+			},
+		}
+		err2 := ioutil.WriteFile(examplePath, []byte(bootstrap1), 0600)
+		Expect(err2).NotTo(HaveOccurred())
+
+		g := NewGenerator(workDir, installerCacheDir, cluster, "", log).(*installerGenerator)
+		err = g.updateBootstrap(examplePath)
+
+		bootstrapBytes, _ := ioutil.ReadFile(examplePath)
+		config, _, _ := config_31.Parse(bootstrapBytes)
```

Comment:
It's possible that a bug could cause the update to produce an invalid ignition. Maybe we should check the error and `report.IsFatal()` here?

Reply:
Definitely! Thanks for the reminder. I'm sure I just breezed past this trying to get tests working, and probably the triple-return-value threw me off of remembering to check it.